searchlib/src/vespa/searchlib/fef/termfieldmatchdata.h

```diff
@@ -66,7 +67,18 @@ class TermFieldMatchData
     size_t capacity() const { return allocated() ? _data._positions._allocated : 1; }
     void reservePositions(size_t sz) {
         if (sz > capacity()) {
-            resizePositionVector(sz);
+            if (allocated()) {
+                resizePositionVector(sz);
+            } else {
+                TermFieldMatchDataPosition * n = new TermFieldMatchDataPosition[sz];
```

Comment:
Consider sharing allocation code with allocateVectorAndAppend() as we are basically doing the same.